packages/hint/docs/contributor-guide/getting-started/release.md

```diff
@@ -0,0 +1,229 @@
+# 🚀 Release process automation and improvements
+
+**STATUS: WORK IN PROGRESS**
+
+## Background
+
+[hint][hint] is a monorepo. This means it contains several packages under the same source control so it is easier to update them. In this case all those packages are under the `packages` folder: main CLI, `hint`s, `connector`s, `formatter`s, `configuration`s, etc. Everything maintained by the team lives there.
+
+Releasing a monorepo following `semver` can be a challenge. How to bump versions? How to deal with circular dependencies?
+
+This document aims to detail what webhint's current release process does, what are the pain points, and a proposal on how to aleviate them.
+
+## Bump versioning using commit tags
+
+Commit messages are used to decide how to bump the package's version:
+
+* `Docs:` no version change, related to `.md` changes
+* `Build:` no version change, related to build and release scripts
+* `Chore:` refactoring, tests, etc. anything not user-facing will be a `patch` (x.x.X)
+* `Upgrade:` dependency updates with no public API changes are a `patch` (x.x.X)
+* `Fix:` bug fixes are a `patch` (x.x.X)
+* `New:` added functionality that does not break previous behavior is a `minor` (x.X.x)
+* `Breaking:` any change to the public API or output (like error messages) is a `major` (X.x.x)
+
+## Automatic bumb versioning of local dependencies
+
+The release process updates local dependencies directly using `Upgrade` in the commit message. The reason is that [no code changes are needed to upgrade thus a `patch`](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api). If any code modification is required it should be done prior running the release script.
+
+**⁉️ NOTE**: Publishing a new version shouldn't be done if the only change since it was las published is an `Upgrade` of dependency(ies).
+
+Let's imagine we have `packageB`, v1.5.0, that depends on `packageA`, v1.0.5. The following are some scenarios of what should be done:
+
+### No public changes
+
+* `packageA` adds support for something new → `New: Support for this new thing`.
+* `packageB` needs `packageA`'s new feature to fix something that was broken → `Fix: This nasty bug`.
+
+The release script will:
+* release `packageA` with a `minor` bump: version 1.1.0.
+* add a new commit to `packageB`: `Upgrade: Bump packageA to 1.1.0`. Because the commit history has `Fix` and `Upgrade`, `packageB` will be bumped to 1.5.1 (`Fix` == `Upgrade`).
+
+Let's use a more concrete example:
+
+`utils-create-server` switches their API to spawn a new process per server and makes all its API `async`. This is a breaking change so it should be noted as `Breaking: Make API async`. The release script will bump it to the next major version, so if previously it was `3.2.1`, it will be update to `4.0.0`. At the same time we do this breaking change we want to update the packages that depend on it:
```

Comment:
so if previously it was `3.2.1`, it will be update to `4.0.0`. ---> should be "updated to"